src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTableUI.java

```diff
@@ -227,6 +231,12 @@ protected void uninstallDefaults() {
             table.setTransferHandler(null);
         }
         SynthContext context = getContext(table, ENABLED);
+        if (showHorizLines) {
```

Comment:
I think we should make this condition tighter to avoid unnecessary repaints like in GTK LAF.
Like only when current property is false and we expect it to be true and then only call setShowHorizontal/VerticalLines()

Reply:
I am not sure what you meant..
uninstallDefaults will be called only once during L&F transition from Nimbus to other L&F and
 it will set horizontal and vertical lines only if it was set in other L&F before Nimbus L&F takes effect..
If other L&F did not have those set, it will not call setShowHorizontal/VerticalLines()

Reply:
Will uninstallDefaults() be called for GTK -> other LAF transition?
I was thinking SynthTableUI is common for all SynthLAF's and if we are transitioning from GTK -> Other LAF's there is no need to call setShowHorizontal/VerticalLines(). Because as mentioned in description only Nimbus LAF will have "showGrid" as false and for GTK we will not be calling `table.setShowGrid(false);`

Reply:
Yes, for GTK showGrid is true so we can add that condition..
Fix updated...